### PR TITLE
Add concurrency for flakey tests detection

### PR DESCRIPTION
**Pull request by ijjk**
This ensures we parallelize our flakey test detection since when a lot are changed it can hit the timeout and not be able to run them all.

### CODE REVIEW

File: scripts/test-new-tests.mjs [hunk 1]
```diff
@@ -102,7 +117,35 @@ async function main() {
...
+    // get the smallest group time to add current one to
+    for (let i = 0; i < groupTotal; i++) {
+      if (!fileGroups[i]) {
+        fileGroups[i] = []
+      }
+
+      if (
+        smallestGroup &&
+        fileGroups[i] &&
+        fileGroups[i].length < smallestGroup?.length
```
Comment (ztanner) [inline scripts/test-new-tests.mjs:140]:
```suggestion
        fileGroups[i].length < smallestGroup.length
```

File: scripts/test-new-tests.mjs [hunk 2]
```diff
@@ -2,9 +2,24 @@
...
 import execa from 'execa'
 import path from 'path'
+import yargs from 'yargs'
 
 async function main() {
-  let testMode = process.argv.includes('--dev-mode') ? 'dev' : 'start'
+  let argv = await yargs(process.argv.slice(2))
+    .boolean('--dev-mode')
+    .string('g')
+    .alias('g', 'group').argv
```
Comment (ztanner) [inline scripts/test-new-tests.mjs:11]: do we need an alias here?

**Review by ztanner — APPROVED** [review]
lgtm, small nits

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 112a34f66aaf996d968fbcbff3188cdfbd05f14b

`TURBOPACK=1 pnpm test-start test/e2e/prerender.test.ts` (turbopack)
- Prerender > should on-demand revalidate for fallback: blocking with onlyGenerated if generated

<details>
<summary>Expand output</summary>

  ● Prerender › should on-demand revalidate for fallback: blocking with onlyGenerated if generated

    expect(received).toMatch(expected)

    Expected pattern: /(HIT|STALE)/
    Received string:  "MISS"

      2300 |
      2301 |         expect(initialTime).toBe($2('#time').text())
    > 2302 |         expect(res2.headers.get('x-nextjs-cache')).toMatch(/(HIT|STALE)/)
           |                                                    ^
      2303 |
      2304 |         const res3 = await fetchViaHTTP(
      2305 |           next.url,

      at Object.toMatch (e2e/prerender.test.ts:2302:52)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Pull request merged**